src/main/java/echo/main/Echo.java

```diff
@@ -0,0 +1,233 @@
+package echo.main;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import echo.exception.EchoException;
+import echo.parser.Parser;
+import echo.storage.Storage;
+import echo.task.Deadlines;
+import echo.task.Events;
+import echo.task.Task;
+import echo.task.ToDos;
+import echo.tasklist.TaskList;
+import echo.ui.Ui;
+
+/**
+ * Represents the Chat Bot called Echo
+ *
+ * @author Ernest Lim
+ */
+public class Echo {
+    private Storage storage;
+    private Ui ui;
+    private TaskList taskList;
+
+    private Parser parser;
+
+    /**
+     * Constructor to create Echo Object
+     *
+     * @param filePath String that contains the file path to store or load the list of task
+     */
+    public Echo(String filePath) {
+        this.ui = new Ui();
+        this.taskList = new TaskList();
+        this.parser = new Parser();
+        this.storage = new Storage(filePath);
+    }
+
+    public String greetUser() {
+        try {
+            this.storage.loadStorage(this.parser, this.taskList);
+            return ui.greet();
+        } catch (IOException e) {
+            return ui.printErrorMessage(e);
+        }
+    }
```

Comment:
Is this method doing more than what its name suggests? You can consider breaking this method into two separate methods.

The method `greetUser()` currently has a side effect of loading file contents into your `taskList` and needs to handle a possible `IOException`—which is thrown from the `this.storage,loadStorage()` method and not the `ui.greet()` method.

This behavior will not be expected when other classes call `greetUser()` as the current method name makes no mention of loading tasks from a file.

You can also consider keeping the current logic as is, but renaming this method to something like `initialise()` to reduce the chances of misunderstandings when reading the code.